Added example with salesforce commerce cloud

Nutlope opened the pull request:
Adding a Next.js example with Salesforce Commerce Cloud

Nutlope commented:
Anyone know how to fix this error?

<img width="687" alt="CleanShot 2022-10-12 at 20 12 30@2x" src="https://user-images.githubusercontent.com/63742054/195470278-2babba15-4563-4a60-a520-74a8065443d5.png">

balazsorban44 commented on examples/with-sfcc/.example.env:
```diff
@@ -0,0 +1,5 @@
+SFDC_CLIENT_ID=
```
This file should be renamed to `.env.local.example` so it won't accidentally be committed.

Nutlope replied:
We want it to be committed though since that's the example env file showing that env variables are needed

manovotny replied:
@Nutlope, @balazsorban44 is saying to jeep the file but rename it to `.env.local.example`. So that when someone does use it and removes the `.example` part, it will say `.env.local`, which is already ignored by `.gitignore`.

balazsorban44 commented on examples/with-sfcc/.gitignore:
```diff
@@ -0,0 +1,38 @@
+yarn-error.log*
+
+# local env files
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+.env
```
```suggestion
.env*.local
```

balazsorban44 commented on examples/with-sfcc/.gitignore:
```diff
@@ -0,0 +1,38 @@
+.vercel
+
+# typescript
+*.tsbuildinfo
```
```suggestion
*.tsbuildinfo
next-env.d.ts
```

balazsorban44 commented on examples/with-sfcc/README.md:
````diff
@@ -0,0 +1,27 @@
+
+Execute [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/with-sfcc) with [npm](https://docs.npmjs.com/cli/init), [Yarn](https://yarnpkg.com/lang/en/docs/cli/create/), or [pnpm](https://pnpm.io) to bootstrap the example::
+
+```bash
+npx create-next-app --example with-sfcc nextjs-sfcc-app
+# or
+yarn create next-app --example with-sfcc nextjs-sfcc-app
+# or
+pnpm create next-app --example with-sfcc nextjs-sfcc-app
+```
````
Should look like this  (cannot suggest this because the code fences also use three "`")

https://github.com/vercel/next.js/blob/562b5a380ead4a41abdb1574e262a8a24bcbe0a3/examples/hello-world/README.md?plain=1#L15-L25

balazsorban44 commented on examples/with-sfcc/components/ProductCard.js:
```diff
@@ -0,0 +1,37 @@
+import Image from 'next/future/image'
+import Link from 'next/link'
+import { useState } from 'react'
+
+export default function ProductCard({ product }) {
+  const [isLoading, setLoading] = useState(true)
+  function cn(...classes) {
+    return classes.filter(Boolean).join(' ')
+  }
```
```suggestion

function cn(...classes) {
  return classes.filter(Boolean).join(' ')
}

export default function ProductCard({ product }) {
  const [isLoading, setLoading] = useState(true)

```

balazsorban44 commented on examples/with-sfcc/package.json:
```diff
@@ -0,0 +1,27 @@
+  "dependencies": {
+    "commerce-sdk": "^2.8.0",
+    "next": "latest",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2"
```
```suggestion
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
```

Could you also verify that all other dependencies are up to date?

balazsorban44 commented on examples/with-sfcc/package.json:
```diff
@@ -0,0 +1,27 @@
+    "@types/node": "17.0.4",
+    "@types/react": "17.0.38",
+    "autoprefixer": "^10.4.0",
+    "eslint": "8.13.0",
+    "eslint-config-next": "12.1.5",
```
```suggestion
```

balazsorban44 commented on examples/with-sfcc/package.json:
```diff
@@ -0,0 +1,27 @@
+    "eslint": "8.13.0",
+    "eslint-config-next": "12.1.5",
+    "postcss": "^8.4.5",
+    "prettier": "^2.5.1",
+    "prettier-plugin-tailwindcss": "^0.1.1",
```
```suggestion
```

balazsorban44 commented on examples/with-sfcc/pages/_document.js:
```diff
@@ -0,0 +1,26 @@
+import Document, { Html, Head, Main, NextScript } from 'next/document'
+
+class MyDocument extends Document {
```
Let's convert to a functional component: https://nextjs.org/docs/advanced-features/custom-document

balazsorban44 commented on examples/with-sfcc/prettier.config.js:
```diff
@@ -0,0 +1,4 @@
+module.exports = {
```
Let's remove this file.

balazsorban44 reviewed (changes requested):
Added some feedback. Could you convert the example to TypeScript? (I also did not see the `tsconfig.json` file, make sure to add it.)

Also, could you go over the criteria here: https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md

manovotny commented:
Excellent work @Nutlope and great suggestions @balazsorban44! 👏

Nutlope commented:
Thanks @manovotny! It was both of us :)

Nutlope commented:
@balazsorban44 Thanks for the detailed feedback! I went over the checklist and addressed all your comments (besides the .env.example which we want to commit). 

I'd love to get this merged in for now and I can come back and convert to TypeScript after Next.js Conf as things are quite busy these next two weeks. cc @manovotny, lemme know if you agree with this thinking

balazsorban44 commented:
Sure, we can follow-up in a later PR :+1:

manovotny commented:
> @balazsorban44 Thanks for the detailed feedback! I went over the checklist and addressed all your comments.
> 
> I'd love to get this merged in for now and I can come back and convert to TypeScript after Next.js Conf as things are quite busy these next two weeks. cc @manovotny, lemme know if you agree with this thinking

I support this iterative approach and commit us to following up, as long as TS isn't a requirement for all examples.
